bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```diff
@@ -0,0 +1,449 @@
+        Optional<Home> optionalHome = model.findHomeByRoomId(roomId);
+        Optional<Room> optionalRoom = model.findRoomById(roomId);
+        if (optionalHome.isPresent() && optionalRoom.isPresent()) {
+            SetRoomTempRequest req = new SetRoomTempRequest(optionalHome.get(), optionalRoom.get());
+            if (command instanceof QuantityType<?>) {
+                int newTemp = (int) ((QuantityType<?>) command).longValue();
+                switch (mode) {
+                    case Sleep:
+                        req.sleepTemp = newTemp;
+                        break;
+                    case Away:
+                        req.awayTemp = newTemp;
+                        break;
+                    case Comfort:
+                        req.comfortTemp = newTemp;
+                        break;
+                    default:
+                        logger.info("Cannot set room temp for mode " + mode);
+                }
+                try {
+                    sendLoggedInRequest(req, SetRoomTempResponse.class);
+                } catch (MillheatCommunicationException e) {
+                    logger.info("Error updating temperature for room {}", roomId, e);
+                }
+            } else {
+                logger.info("Error updating temperature for room {}, expected QuantityType but got {}", roomId,
+                        command);
+            }
+        }
+    }
+
+    public void updateIndependentHeaterProperties(@Nullable String macAddress, @Nullable String heaterId,
+            @Nullable Command temperatureCommand, @Nullable Command masterOnOffCommand, @Nullable Command fanCommand) {
+        Optional<Heater> optionalHeater = model.findHeaterByMacOrId(macAddress, heaterId);
+        if (optionalHeater.isPresent()) {
+            Heater heater = optionalHeater.get();
+            int setTemp = heater.getTargetTemp();
+            if (temperatureCommand != null && temperatureCommand instanceof QuantityType<?>) {
+                setTemp = (int) ((QuantityType<?>) temperatureCommand).longValue();
+            }
+            boolean masterOnOff = heater.isPowerStatus();
+            if (masterOnOffCommand != null) {
+                masterOnOff = masterOnOffCommand == OnOffType.ON ? true : false;
+            }
+            boolean fanActive = heater.isFanActive();
+            if (fanCommand != null) {
+                fanActive = fanCommand == OnOffType.ON ? true : false;
+            }
+            SetDeviceTempRequest req = new SetDeviceTempRequest(heater, setTemp, masterOnOff, fanActive);
+            try {
+                sendLoggedInRequest(req, SetRoomTempResponse.class);
+                heater.setTargetTemp(setTemp);
+                heater.setPowerStatus(masterOnOff);
+                heater.setFanActive(fanActive);
+            } catch (MillheatCommunicationException e) {
+                logger.info("Error updating temperature for heater {}", macAddress, e);
```

Comment:
Is logging this enough, should you not inform the caller that you failed to update...